app/uk/gov/hmrc/economiccrimelevyregistration/models/EclRegistrationModel.scala

```diff
@@ -19,5 +19,6 @@ package uk.gov.hmrc.economiccrimelevyregistration.models
 case class EclRegistrationModel(
   registration: Registration,
   registrationAdditionalInfo: Option[RegistrationAdditionalInfo] = None,
-  hasRegistrationChanged: Boolean = false
+  hasRegistrationChanged: Boolean = false,
+  additionalInfoChanged: Boolean = false
```

Comment:
Should this be a field in the EclRegistrationModel? Can we not set it as a value instead in the class

```
val answerChanged = request.additionalInfo match {
               case None       => true
               case Some(info) => !info.liableForPreviousYears.contains(liableBeforeCurrentYear)
             }
```

Reply:
No because the way these `boolean`s are worked out depends on what bit of the actual case class has changed and only the controllers' `onSubmit` know this.